src/test/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtilsTest.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.testFramework.fixtures.BasePlatformTestCase;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+/**
+ * Tests for {@link LSPIJEditorUtils}.
+ */
+public class LSPIJEditorUtilsTest extends BasePlatformTestCase {
+
+    public void testTextMateQuoteCharactersForTypeScript() {
```

Comment:
Added tests to confirm that `LSPIJEditorUtils#getQuoteCharacters()` now uses TextMate metadata to determine which subset of quote characters are supported for a given TextMate file. I selected two file extensions that are pre-registered as enabled TextMate bundles in IJ CE and that have different sets of quote characters, namely TypeScript which has single-quoted strings, double-quoted strings, and backtick-quoted strings and PHP which only has single-quoted strings and double-quoted strings.